service/src/test/java/org/broadinstitute/listener/relay/http/RelayedHttpRequestProcessorTest.java

```diff
@@ -172,9 +173,14 @@ void writeTargetResponseOnCaller_responseIsWrittenBackToCaller() throws IOExcept
 
     processor.writeTargetResponseOnCaller(targetHttpResponse);
 
-    verify(responseStream).write(responseData.capture());
+    // writeTargetResponseOnCaller relies on StreamUtils.copy, which specifies
+    // an offset and length in its call to write()
+    verify(responseStream).write(responseData.capture(), anyInt(), anyInt());
 
-    assertThat(new String(responseData.getValue()), equalTo(BODY_CONTENT));
+    // StreamUtils.copy buffers its writes in arrays of 4096 bytes (by default). Since this
```

Comment:
Naive question, but would it be possible / useful to set up a test response that is on the order of size of what triggered the OOM?

Reply:
added new unit tests for this.